src/backend/api/meals.js

```diff
@@ -2,11 +2,190 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+const mealsColumns = new Set([
+  "title",
+  "description",
+  "location",
+  "when",
+  "max_reservations",
+  "price",
+  "created_date",
+]);
+
+// this function checks if input data valid and makes possible to update only neccesary data in table
+function checkTableData(table) {
+  try {
+    let validTableData = {};
+    for (const key in table) {
+      if (mealsColumns.has(key)) {
+        validTableData[key] = table[key];
+      }
+    }
+    if ("when" in validTableData) {
+      const whenFormatted = new Date(validTableData.when);
+      if (!Date.parse(whenFormatted)) {
+        throw new Error();
+      }
+    }
+    if ("max_reservations" in validTableData) {
+      if (isNaN(Number(validTableData.max_reservations))) {
+        throw new Error();
+      }
+    }
+    if ("price" in validTableData) {
+      if (isNaN(Number(validTableData.price))) {
+        throw new Error();
+      }
+    }
+    if ("created_date" in validTableData) {
+      const createdDateFormatted = new Date(validTableData.created_date);
+      if (!Date.parse(createdDateFormatted)) {
+        throw new Error();
+      }
+    }
+    return validTableData;
+  } catch (error) {
+    throw error;
+  }
+}
+
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const maxPrice = request.query.maxPrice;
+    const title = request.query.title;
+    const availableReservations = request.query.availableReservations;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    let meals = knex("meals");
+    let filteredMeals;
+    if (Object.keys(request.query).length === 0) {
+      response.json(await meals);
+    } else {
+      if (maxPrice) {
```

Comment:
I'm not sure if this function will work as you expect. Since you add `await` to each knex part you will actually execute the query and overwrite the value from the last query, rather than build on to that query.

Further, it would be better if you handled all of the error checking up front, by either throwing or using some default value instead.

Then write the query as if all the parameters are set but as a single flow, not based on all these conditions :)